### ISSUE TITLE
[The Longest Journey] Audio issues...

### ISSUE THREAD

**Issue by joveian** [residualvm/residualvm#1532]
With ResidualVM 0.4.0git439-g5c24080e (Mar 1 2019 10:44:38) and the GOG version of The Longest Journey on Window 10 audio is frequently but not always garbled.  Looking at a recording via audacity WASAPI, there is obviously no audio being output at times when it should.  I am attaching a brief clip showing the issue.  The only other game I have notice this with is RiME (that makes heavy use of both CPU and GPU); a number of others have worked without issue.

The first dialog of the game shows this until the point April is saying "Oh perfect", which is fine.  I'd guess this affects maybe a quarter of the dialog throughout the game.  The cave on the island (the one you need to use the rope to get into) was the worst.  I normally had subtitles on, but tested it quick without and it seems to have the same issue.

Since this is an offline Windows 10 system it hasn't been fully up to date (but is fairly recent with updates mid or late December) and I was using the default audio driver from windows install (chip is Realtek ALC1220).  I quickly tried the drivers from the motherboard manufacturer's website and they are at least as bad.  I also tried it with USB audio and that also shows the issue.

[tlj-audio-issue.zip](https://github.com/residualvm/residualvm/files/3027601/tlj-audio-issue.zip)

**Comment by bgK** [comment]
Could try replacing the SDL2.dll that comes with the daily build with version 2.0.9 from there and see if it fixes the issue: https://www.libsdl.org/release/SDL2-2.0.9-win32-x86.zip
(And perhaps #1530 as well)

**Comment by joveian** [comment]
With that change it doesn't completely fail to output audio (and doesn't sound nearly as harsh) but it still doesn't output correct audio and is garbled in the same places.  It doesn't have any effect that I can see on the font issue in #1530.

Here is the same clip with the new SDL2.dll:

[tlj-audio-issue-sd2-2.0.9.zip](https://github.com/residualvm/residualvm/files/3060091/tlj-audio-issue-sd2-2.0.9.zip)

**Comment by joveian** [comment]
I tried several things and they are all worse than the default, hopefully in interesting ways.  output_rate=48000 causes more frequent gaps in the audio output at regular intervals (clip attached).  Larger buffers cause longer pauses and the animation also seems to pause at  the same time.  I've attached a clip with audio_buffer_size=16384, but even 1024 seems worse than the default (which I think might be 512).  I tried lower powers of two down to 4 and there isn't much difference below 512 so either there is a minimum I am hitting or it just doesn't matter.  Also, I was wrong last time and it does still not output audio with SDL2 2.0.9, the gaps are just smaller and I didn't notice them in the default audacity zoom.

There does seem to be some variation where exactly the audio clears up, even with the same settings (possibly related to exactly when I skipped previously).  It usually clears up sometime during the line after "oh...perfect", but sometimes that is still garbled and looking at the tree is ok, or the first look at the tree is still garbled but the second is ok.  The lower buffer sizes both sound better and clear up faster.  With the higher buffer settings it doesn't ever fully clear up, although it does still get much better.

One thing I am wondering is if mixing sound has something to do with it.  It doesn't seem like a perfect correlation but it does seem like music or sound in the background (such as the firepace in the initial dialog) make it much more likely to have trouble.

[tlj-audio-issue-48k.zip](https://github.com/residualvm/residualvm/files/3061012/tlj-audio-issue-48k.zip)

[tlj-audio-issue-buffer-16384.zip](https://github.com/residualvm/residualvm/files/3061013/tlj-audio-issue-buffer-16384.zip)
